### DISCUSSION TITLE
Relationship between disk reads and grains

### DISCUSSION THREAD

**Discussion by mshallo** [dotnet/orleans#7795] [General]
Hi All,

When reviewing the Scaleset instances where our Silo's live. I notice a pattern with the Disk Reads in Bytes. It looks like as grains are activated the state might be stored on Disk (Virtual memory) to some extent rather than in the RAM itself.

We sized our vm's to be Memory focused so I find the disk read spikes unusual.

Could I get some clarification on what the relationship is between Orleans and its use of the VM's Disk.

Thanks,
Matt

**Comment by ReubenBond** [comment]
Hi @mshallo, I completely missed this discussion topic - my apologies. Orleans does not use the local disk at all. I wonder if you're seeing something pagefile/swap related. If that's the case, then it may be worth collecting a memory dump. Perhaps there's a memory leak in the application (or Orleans). Either way, we will help you to work through it.
